Source/JavaScriptCore/bytecode/InlineCacheCompiler.cpp

```diff
@@ -226,6 +226,8 @@ auto InlineCacheCompiler::preserveLiveRegistersToStackForCall(const RegisterSet&
 
     unsigned extraStackPadding = 0;
     unsigned numberOfStackBytesUsedForRegisterPreservation = ScratchRegisterAllocator::preserveRegistersToStackForCall(*m_jit, liveRegisters.buildAndValidate(), extraStackPadding);
+    RELEASE_ASSERT(liveRegisters.buildAndValidate().numberOfSetRegisters() == liveRegisters.buildScalarRegisterSet().numberOfSetRegisters());
```

Comment:
Do you know that you can specify more arguments to the RELEASE_ASSERT e.g.
```
RELEASE_ASSERT(liveRegisters.buildAndValidate().numberOfSetRegisters() == liveRegisters.buildScalarRegisterSet().numberOfSetRegisters(), liveRegisters.buildAndValidate().numberOfSetRegisters(), liveRegisters.buildScalarRegisterSet().numberOfSetRegisters());
```

That way, if the RELEASE_ASSERT fails, those values will be captured in the register file for inspection.  Just don't capture any PII.  I suggest you do the same for all the other RELEASE_ASSERTs too that you're adding for this diagnostic.